typescript/xstate/src/lib.ts

```diff
@@ -0,0 +1,360 @@
+    start: () => {
+    }
+  };
+
+  return system;
+}
+
+interface FakeParent<TLogic extends AnyStateMachine> extends AnyActorRef {
+  _send: (event: EventFromLogic<TLogic>) => void;
+}
+
+export async function createActor<TLogic extends AnyStateMachine>(ctx: restate.RpcContext, api: XStateApi<TLogic>, systemName: string, logic: TLogic extends AnyStateMachine ? AreAllImplementationsAssumedToBeProvided<TLogic['__TResolvedTypesMeta']> extends true ? TLogic : MissingImplementationsError<TLogic['__TResolvedTypesMeta']> : TLogic, options?: ActorOptions<TLogic>): Promise<Actor<TLogic>> {
+  const system = await createSystem(ctx, api, systemName)
+  const snapshot = await ctx.get<Snapshot<unknown>>("snapshot") ?? undefined;
+
+  const parent: FakeParent<TLogic> = {
+    id: "fakeRoot",
+    sessionId: "fakeRoot",
+    send: () => {
+    },
+    _send: () => {
+    },
+    start: () => {
+    },
+    getSnapshot: (): null => {
+      return null
+    }, // TODO
+    getPersistedSnapshot: (): Snapshot<unknown> => {
+      return {
+        status: 'active',
+        output: undefined,
+        error: undefined,
+      }
+    }, // TODO
+    stop: () => {
+    }, // TODO
+    system,
+    src: "fakeRoot",
+    subscribe: (): Subscription => {
+      return {
+        unsubscribe() {
+        }
+      }
+    },
+    [Symbol.observable]: (): InteropSubscribable<any> => {
+      return {
+        subscribe(): Subscription {
+          return {
+            unsubscribe() {
+            }
+          }
+        }
+      }
+    },
+  }
+
+  if (options?.inspect) {
+    // Always inspect at the system-level
+    system.inspect(toObserver(options.inspect));
+  }
+
+  return createXActor(logic, {id: "root", ...options, parent, snapshot} as any);
+}
+
+const actorMethods = <TLogic extends AnyStateMachine>(path: string, logic: TLogic) => {
+  const api = xStateApi(path)
+
+  return {
+    create: async (ctx: restate.RpcContext, systemName: string, request?: { input?: InputFrom<TLogic> }): Promise<Snapshot<unknown>> => {
+      ctx.clear("snapshot")
+      ctx.clear("events")
+      ctx.clear("children")
+
+      const root = (await createActor(ctx, api, systemName, logic, {
+        input: request?.input,
+      })).start();
+
+      ctx.set("snapshot", root.getPersistedSnapshot())
+
+      return root.getPersistedSnapshot()
+    },
+    send: async (ctx: restate.RpcContext, systemName: string, request?: { scheduledEvent?: SerialisableScheduledEvent, source?: SerialisableActorRef, target?: SerialisableActorRef, event: AnyEventObject }): Promise<Snapshot<unknown> | undefined> => {
+      if (!request) {
+        throw new TerminalError("Must provide a request")
+      }
+
+      if (request.scheduledEvent) {
+        const events = await ctx.get<{ [key: string]: SerialisableScheduledEvent }>("events") ?? {}
+        const scheduledEventId = createScheduledEventId(request.scheduledEvent.source, request.scheduledEvent.id)
+        if (!(scheduledEventId in events)) {
```

Comment:
Restate doesn't yet support directly cancellable delayed calls, so I sort of implement it myself by just checking to see if the event is still scheduled, and also that it hasn't been replaced